src/net_processing.cpp

```diff
@@ -3599,10 +3705,12 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         {
         LOCK(cs_main);
 
-        if (!m_chainman.m_blockman.LookupBlockIndex(cmpctblock.header.hashPrevBlock)) {
+        if (!LookupBlockIndexForPeer(pfrom, cmpctblock.header.hashPrevBlock)) {
             // Doesn't connect (or is genesis), instead of DoSing in AcceptBlockHeader, request deeper headers
-            if (!m_chainman.ActiveChainstate().IsInitialBlockDownload())
+            if (!m_chainman.ActiveChainstate().IsInitialBlockDownload()) {
                 m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::GETHEADERS, m_chainman.ActiveChain().GetLocator(m_chainman.m_best_header), uint256()));
+                UpdateChainTipSet(GetChainTipSetForPeerNetwork(pfrom), m_chainman.m_best_header);
```

Comment:
Are we calling `UpdateChainTipSet()` any time a getheaders message is sent? If so, what do you think about factoring that out into a function, so that if more `PushMessage(getheaders)` are added, people don't forget to add `UpdateChainTipSet()`?

Reply:
Seemed easy enough, so i included it here.